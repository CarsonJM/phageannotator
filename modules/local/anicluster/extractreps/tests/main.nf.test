nextflow_process {

    name "Test process: ANICLUSTER_EXTRACTREPS"
    script "../main.nf"
    process "ANICLUSTER_EXTRACTREPS"
    tag "modules"
    tag "modules_local"
    tag "anicluster"
    tag "anicluster_extractreps"


    test("['bacteroides_fragilis']['illumina']['test1_contigs_fa_gz'] & ['modules_local']['clusters_tsv']") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id: 'test' ], [ file(params.test_data['bacteroides_fragilis']['illumina']['test1_contigs_fa_gz'], checkIfExists: true ) ],
                                                [ file(params.test_data['modules_local']['clusters_tsv'], checkIfExists: true ) ] ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("['bacteroides_fragilis']['illumina']['test1_contigs_fa_gz'] & ['modules_local']['clusters_tsv'] - stub") {

        options "-stub"

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id: 'test' ], [ file(params.test_data['bacteroides_fragilis']['illumina']['test1_contigs_fa_gz'], checkIfExists: true ) ],
                                                [ file(params.test_data['modules_local']['clusters_tsv'], checkIfExists: true ) ] ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
