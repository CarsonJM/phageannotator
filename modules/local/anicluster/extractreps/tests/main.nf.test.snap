{
    "['bacteroides_fragilis']['illumina']['contigs_fa_gz'] & ['modules_local']['clusters_tsv']": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_anicluster_representatives.fna.gz:md5,b4ae3ed3aa26b50c3e3d3277e94ac940"
                    ]
                ],
                "1": [
                    "versions.yml:md5,15b724eac7b61bbed8de17f17eae0bf8"
                ],
                "representatives": [
                    [
                        {
                            "id": "test"
                        },
                        "test_anicluster_representatives.fna.gz:md5,b4ae3ed3aa26b50c3e3d3277e94ac940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,15b724eac7b61bbed8de17f17eae0bf8"
                ]
            }
        ],
        "timestamp": "2023-12-06T15:24:32.364815085"
    },
    "['bacteroides_fragilis']['illumina']['contigs_fa_gz'] & ['modules_local']['clusters_tsv'] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_anicluster_representatives.fna.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,15b724eac7b61bbed8de17f17eae0bf8"
                ],
                "representatives": [
                    [
                        {
                            "id": "test"
                        },
                        "test_anicluster_representatives.fna.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,15b724eac7b61bbed8de17f17eae0bf8"
                ]
            }
        ],
        "timestamp": "2024-01-17T22:49:59.94544688"
    }
}